src/components/Airtable.js

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import {
+  InternalLinks,
+  AIRTABLE_ASSETS_RULES,
+  AIRTABLE_CATEGORIES_RULES,
+  PULL_ASSETS_TIME,
+  PULL_CATEGORIES_TIME,
+  verifyDataAirtable,
+} from '../constants';
+
+export const AirtableContext = React.createContext(null);
+
+class AirtableProvider extends React.Component {
+  constructor(props){
+    super(props);
+    this.getCategoriesForAssets = this.getCategoriesForAssets;
+    this.state = {
+      getCategoriesForAssets: this.getCategoriesForAssets,
+    }
+  }
+
+  componentDidMount = () => {
+    this.getAssets();
+    this.getCategories();
+    this.setIntervals();
+  }
+
+  componentWillUnmount = () => {
+    this.resetIntervals();
+  }
+
+  setIntervals = () => {
+    this.intervalPullAssets = setInterval(this.getAssets, PULL_ASSETS_TIME)
+    this.intervalPullCategories = setInterval(this.getCategories, PULL_CATEGORIES_TIME)
+  }
+
+  resetIntervals = () => {
+    clearInterval(this.intervalPullAssets);
+    clearInterval(this.intervalPullCategories);
+  }
+
+  processAssetsFromAirTable = ({ fields }) => {
+    let location = undefined;
+    if(fields.Location){
+      let countries = fields.Location.split(',');
+      location = {};
+      countries.forEach(country => {
+        country = country.trim();
+        let cities = /\(([^)]+)\)/g.exec(country);
+
+        if(cities){
+          country = country.substring(0, country.indexOf('('))
+          cities = cities[1].split(';');
+          location[country] = cities;
+        } else {
+           location[country] = {};
```

Comment:
I learned a new pattern to handle these kind of use cases in a more functional way, let me know what you think. Instead of `if else` this, how about we do the following:
```
((countries) => {
  countries.map( country => {
    ((cities) => { 
      ...
    })(/\(([^)]+)\)/g.exec(country))
  })
})(fields.Location.split(','))
```
The idea is that you wrap logic in pure functions to try to reduce side effects and memory leaks/reusage. Trying to hold the state of data with arrays and `let` variables is very prone error and hard to maintain.